lib/pay/stripe.rb

```diff
@@ -117,6 +123,29 @@ def self.configure_webhooks
         # Handle subscriptions in Stripe Checkout Sessions
         events.subscribe "stripe.checkout.session.completed", Pay::Stripe::Webhooks::CheckoutSessionCompleted.new
         events.subscribe "stripe.checkout.session.async_payment_succeeded", Pay::Stripe::Webhooks::CheckoutSessionAsyncPaymentSucceeded.new
+
+        events.subscribe "stripe.invoice.upcoming", Pay::Stripe::Webhooks::SubscriptionRenewing.new
+        # Payment action is required to process an invoice
+        events.subscribe "stripe.invoice.payment_action_required", Pay::Stripe::Webhooks::PaymentActionRequired.new
+        # If an invoice payment fails, we want to notify the user via email to update their payment details
+        events.subscribe "stripe.invoice.payment_failed", Pay::Stripe::Webhooks::PaymentFailed.new
+
+        events.subscribe "stripe.invoice.created", Pay::Stripe::Webhooks::InvoiceCreated.new
+        events.subscribe "stripe.invoice.deleted", Pay::Stripe::Webhooks::InvoiceDeleted.new
+        events.subscribe "stripe.invoice.finalization_failed", Pay::Stripe::Webhooks::InvoiceFinalizationFailed.new
+        events.subscribe "stripe.invoice.finalized", Pay::Stripe::Webhooks::InvoiceFinalized.new
+        events.subscribe "stripe.invoice.overdue", Pay::Stripe::Webhooks::InvoiceOverdue.new
+        events.subscribe "stripe.invoice.paid", Pay::Stripe::Webhooks::InvoicePaid.new
+        events.subscribe "stripe.invoice.payment_succeeded", Pay::Stripe::Webhooks::InvoicePaymentSucceeded.new
+        # UPCOMING HAS NO ID
```

Comment:
really? 👀 when is this fired?

Reply:
https://docs.stripe.com/billing/subscriptions/webhooks

`Sent a few days prior to the renewal of the subscription. The number of days is based on the number set for Upcoming renewal events in the [Dashboard](https://dashboard.stripe.com/settings/billing/automatic). For existing subscriptions, changing the number of days takes effect on the next billing period. You can still add [extra invoice items](https://docs.stripe.com/billing/invoices/subscription#adding-upcoming-invoice-items), if needed."`